crates/uv/src/commands/project/sync.rs

```diff
@@ -809,6 +813,54 @@ pub(super) async fn do_sync(
     Ok(())
 }
 
+/// Apply runtime constraints to the extra build requirements.
+#[allow(clippy::result_large_err)]
+fn apply_match_runtime(
+    extra_build_requires: ExtraBuildRequires,
+    resolution: &Resolution,
+) -> Result<ExtraBuildRequires, ProjectError> {
```

Comment:
It feels a little weird that this isn't a method on `ExtraBuildRequires` are you avoiding some circular dependency?

Reply:
I'm just matching the style of the rest of the file, but I suspect it could be a method on `ExtraBuildRequires` yeah.